File: src/content_manager/UUID.zig
const std = @import("std");
const inn = @cImport(@cInclude("UUID/UUID.h"));

const UUIDerr = error{
    ErrorUUID,
};

pub const len = 38;

pub fn createNewUUID(uuidStr: [*]u8) !void {
    if (!inn.createNewUUID(@ptrCast(uuidStr))) return UUIDerr.ErrorUUID;
}

File: src/content_manager/UUID.c
#include "UUID/UUID.h"
#include <windows.h>
#include <rpcdce.h>

bool createNewUUID(unsigned char * uuidStr)
{
    RPC_STATUS status = 0;
    UUID uuid[1];
    RPC_CSTR str = NULL;
    status = UuidCreate(uuid);
    if (status != RPC_S_OK) return false;

    status = UuidToString(uuid, &str);
    if (status != RPC_S_OK) return false;

    memcpy(uuidStr, str, 38);

    (void)RpcStringFree(&str);

    return true;
}
